### PR TITLE
XM Cloud tutorial: data sources

### PR DESCRIPTION
**Pull request by jst-cyr**
## Description / Motivation
As part of ongoing work to build out an XM Cloud tutorial flow, this tutorial introduces the concept of defining templates and mapping them to components.

This will need to be linked to other tutorials in the series when they are available.

## How Has This Been Tested?
Local and [on Vercel](https://developer-portal-git-xm-cloud-a5ce40-sitecoretechnicalmarketing.vercel.app/learn/getting-started/xm-cloud/tutorials/datasources)

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Documentation update (non-breaking change; modified files are limited to the `/data` directory or other markdown files)

## Checklist:
- [X] I have read the Contributing guide.
- [X] My code/comments/docs fully adhere to the Code of Conduct.
- [ ] My change is a code change.
- [X] My change is a documentation change and there are NO other updates required.
- [ ] My change has new or updated images which are stored in the `/public/images` folder that need to be migrated to Sitecore DAM

### CODE REVIEW

**Comment by jst-cyr** [comment]
@lovesitecore : This one has an initial untested draft, no screenshots. I'd like to get your initial review before we go too much further.

Known remaining items:
1. Steps for how to insert an HTML Block (line 416)
2. Template or Template Folder for the Text Teaser Folder template
3. Screenshots

**Comment by jst-cyr** [comment]
Only remaining task is to add screenshots.

**Review by lovesitecore — COMMENTED** [review]
Have a couple of comments

File: apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+1. Click on the **Next** button to advance to the **Path** configuration step. This is where you will map the fields from the template to the visual element.
+1. Select the **Headline** field to map the data from that Text Teaser field to the heading element in the component.
+
+   `INSERT SCREENSHOT OF MAPPING`
+
+1. Click **Complete** to finish the mapping process.
+
+### Map component sub-heading element to Subheadline text field
+
+You will now repeat the process for the the subtitle/subheading component. You will select the subheading visual element and map that to the Text field "Subheadline" in the Text Teaser template
```
[the remark this answers is not in the document]
Comment (jst-cyr) [reply apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md:380]: Agreed with Sebastian to replace the word 'component' with 'element'.

File: apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+
+In order to complete the following tutorial, you will need these resouces:
+
+1. A valid organization login for the [Sitecore Cloud portal](https://portal.sitecorecloud.io)
+1. An existing XM Cloud environment and site created in [Tutorial: How to Set Up and Utilize XM Cloud](setup-xm-cloud)
+1. The Text Teaser component created in [Tutorial: Create a Component using XM Cloud Component Builder](create-a-component)
+
+## Create the module
+
+The very first step will be to create a headless module. Modules are used to organize content in a consistent structure so it can be re-used by the marketing team. Think of a module like an npm package for content. Defining modules makes it very easy for you to make your package/module available to multiple sites and teams. An author will be able to 'install' your module and use it in a consistent way on any site they choose.
```
[the remark this answers is not in the document]
Comment (jst-cyr) [reply apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md:49]: This is resolved in the next commit.

File: apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+   </Alert>
+1. Right-click on the **Project** item in the tree to pop open the context menu again.
+1. In the context menu select **Insert** and then **Headless Module** to create the new module.
+
+   `INSERT SCREENSHOT OF INSERT OPTIONS`
+
+1. Configure the Headless Module settings:
+
+   - **Module name:** Basic Components
+   - **Add to module group:** `sitecore/system/settings/Company Dev`
```
[the remark this answers is not in the document]
Comment (jst-cyr) [reply apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md:82]: Fixed in next commit!

File: apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md [hunk 4]
```diff
@@ -0,0 +1,486 @@
...
+   - **System areas for container folders:** Templates, Branch Templates, Renderings, and Placeholder Settings
```
[the remark this answers is not in the document]
Comment (jst-cyr) [reply apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md:83]: Resolved in the next commit.

File: apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md [hunk 5]
```diff
@@ -0,0 +1,486 @@
...
+   - **Module scaffolding actions:** Headless Site Setup
```
[the remark this answers is not in the document]
Comment (jst-cyr) [reply apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md:84]: New step added after to press the proceed button.

File: apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md [hunk 6]
```diff
@@ -0,0 +1,486 @@
...
+   <AlertIcon />
+   When configuring scaffolding actions, you'll notice an option for Headless Tenant Setup. Tenants are an older terminology. These are now called Site Collections.
+   </Alert>
+
+## Creating the templates
+
+The Headless Module has now been created. You can now check that the required folders exist and start creating the required templates. You will create two templates:
+
+- **Text Teaser** to hold the data from the author, and
+- **Text Teaser Folder** to hold created Text Teaser content items.
```
[the remark this answers is not in the document]
Comment (jst-cyr) [reply apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md:98]: Fixed in next commit.

File: apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md [hunk 7]
```diff
@@ -0,0 +1,486 @@
...
+
+   `SCREENSHOT OF COMPONENTS DATA SOURCES PAGE`
+
+1. Scroll down through the list of data sources and find the one named **Text Teaser**
+
+   `SCREENSHOT OF TEXT TEASER DATA SOURCE`
+
+### Map component heading element to Headline text field
+
+The component that was created in the previous tutorials uses static data in XM Cloud Components. As part of using the new data template, you need to map this Text Teaser data source to the visual elements in XM Cloud Components. You are going to start by mapping the headline text field.
```
[the remark this answers is not in the document]
Comment (jst-cyr) [reply apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md:337]: Good idea! I've added a link, even though that tutorial doesn't exist in this branch. The link should work when we start merging the PRs together into main.

File: apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jst-cyr) [reply apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md:380]: Fixed in the next commit.

File: apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md [hunk 8]
```diff
@@ -0,0 +1,486 @@
...
+   `SCREENSHOT OF SUBHEADLINE EDITING MODE`
+
+   <Alert status="info">
+   <AlertIcon />
+   The pane on the right updates and you can see the **Text** part in the **Content** area which is where you need to go next.
+   </Alert>
+
+1. On the right side panel, click on the **Text** option. This will show that the text is currently **Static**
+1. Change the mode from **Static** to **Mapped**. This will show you the source mapping options.
+1. Find and select the **Text Teaser** source in the list of available sources.
```
[the remark this answers is not in the document]
Comment (jst-cyr) [reply apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md:398]: Line removed in the next commit

File: apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md [hunk 9]
```diff
@@ -0,0 +1,486 @@
...
+1. Click on the blank component to select it.
+1. In the context menu above the component, click on the **Delete** action button. This will remove the current version of the component from the page.
+
+   `INSERT SCREENSHOT OF CONTEXT BAR WITH DELETE BUTTON HOVERED OVER BY MOUSE`
+
+1. In the left page, scroll down the **Components** listing to find the **Text Teaser** component.
+
+   `INSERT SCREENSHOT OF COMPONENTS PANE`
+
+1. Drag the Text Teaser card onto the Pages canvas to add the component to the page. A dialog will display prompting you to assign a content item (data source) to the component.
```
[the remark this answers is not in the document]
Comment (jst-cyr) [reply apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md:460]: Fixed in next commit

File: apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md [hunk 10]
```diff
@@ -0,0 +1,489 @@
...
+1. To begin, open the Sitecore Cloud Portal (https://portal.sitecorecloud.io) and log in.
+1. From your dashboard, select the XM Cloud environment where you have your Text Teaser Component created.
+
+   `INSERT SCREENSHOT OF XM CLOUD APP BEING SELECTED FROM DASHBOARD`
+
+1. From the app context flyout, select **Content Editor** to launch the XM Cloud content tree editing interface.
+
+   `INSERT SCREENSHOT OF XM CLOUD APP CONTEXT FLYOUT`
+
+1. Navigate to: `System/Settings/Project` in the content tree.
```
[the remark this answers is not in the document]
Comment (jst-cyr) [reply apps/devportal/data/markdown/pages/learn/getting-started/xm-cloud/tutorials/datasources.md:60]: Good catch, fixed in next commit!

**Review by lovesitecore — APPROVED** [review]

**Pull request merged**